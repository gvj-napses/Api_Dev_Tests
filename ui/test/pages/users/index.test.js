import { render, screen, waitFor } from "@testing-library/react";
import Users from "../../../pages/users/index";
import * as UserService from "../../../services/user.service"

import sinon from "sinon"

describe("users index", () => {
    const sandbox = sinon.createSandbox();
    beforeEach(() => {

    });

    it("should get users", async () => {
        const users = [{
            id: 1,
            name: 'username'
        }]

        sandbox.mock(UserService).expects('getUsers').once()
            .returns({ status: "success", entity: users });

        render(<Users />);

        const list = screen.getByRole("list", { name: /users/i })
        expect(list).toMatchSnapshot()
        expect(screen.getByText("Create")).toBeInTheDocument();
        waitFor(() => { })
    })

    afterEach(() => {
        sandbox.verifyAndRestore();
    });
})
