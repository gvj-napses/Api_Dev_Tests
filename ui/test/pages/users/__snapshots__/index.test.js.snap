// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users index should get users 1`] = `
<ul
  aria-label="users"
>
  <li>
    username
  </li>
</ul>
`;
